doc/howto/write_an_alphabet/dna2_only_rank.cpp

```diff
@@ -0,0 +1,26 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+//! [struct]
+#include <seqan3/alphabet/concept.hpp>                   // alphabet concept checks
+
+struct dna2
+{
+    uint8_t rank{};
```

Comment:
We are speaking about a numerical representation and although bool would be the most memory-efficient type it might be somehow unintuitive for the user...

Reply:
I am not saying it has to be bool, but there should be some guide for users which type to chose. bool is in fact not more memory efficient than `uint8_t`.